[jamesberyl] iP

jamesberyl opened the pull request:
# Nimbus

> “The secret of getting ahead is getting started.” – Mark Twain

Nimbus frees your mind of having to remember things you need to do. It's:

- **user-friendly**
- *cross-platform* compatible
- ~~slow~~ **SUPER FAST** 🚀 to use

All you need to do is,

1. download it from [here](https://github.com/your-repo/nimbus).
2. install it on your device.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Integration with calendar apps (coming soon)
- [ ] Smart reminders for important deadlines (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Nimbus {
    public static void main(String[] args) {
        System.out.println("Welcome to Nimbus!");
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

jamesberyl committed:
Add nimbus

jamesberyl committed:
Improve skeleton of Nimbus

jamesberyl committed:
Added new ability and personality

jamesberyl committed:
Added Task class

jamesberyl committed:
Added Task class

jamesberyl committed:
Added ability to mark tasks

jamesberyl committed:
Modified response

jamesberyl committed:
Added subclasses of Task

jamesberyl committed:
Add support for tracking three types of tasks

jamesberyl committed:
testUI

jamesberyl committed:
Create Nimbus exception class

jamesberyl committed:
Handle errors

jamesberyl committed:
Add support for Delete

jamesberyl committed:
testUI

jamesberyl committed:
Add enumeration

jamesberyl committed:
Add storage functionality

jamesberyl committed:
Merge branch-Level-7 into master

jamesberyl committed:
Add date functionality

jamesberyl committed:
Merge branch-Level-8 into master

jamesberyl committed:
Add new test inputs

jamesberyl committed:
.gitignore: Add text-ui-test/data/nimbus.txt

jamesberyl committed:
Fix error message for find_date

jamesberyl committed:
Add Clear command

jamesberyl committed:
Add new test inputs

jamesberyl committed:
Add confirmation for clear command

jamesberyl committed:
Fix find_date command

jamesberyl committed:
Modify find_date command

jamesberyl committed:
Modify expected test input and outputs

jamesberyl committed:
Modify Deadline class

jamesberyl committed:
Modify Event class

jamesberyl committed:
Add Parser class

jamesberyl committed:
Add TaskList class

jamesberyl committed:
Add UI class

jamesberyl committed:
Update Storage class

jamesberyl committed:
Update Nimbus class

jamesberyl committed:
Add packages

jamesberyl committed:
Update runtest script

jamesberyl committed:
Merge add-gradle-support into master

jamesberyl committed:
Update build.gradle

jamesberyl committed:
Update packages

jamesberyl committed:
Add JUnit tests

jamesberyl committed:
Add JavaDoc

jamesberyl committed:
Apply coding standards

jamesberyl committed:
Add find functionality

jamesberyl committed:
Merge branch-A-JavaDoc into master

jamesberyl committed:
Merge branch-A-CodingStandard into master

jamesberyl committed:
Merge branch-Level-9 into master

# Conflicts:
#	src/main/java/nimbus/parser/Parser.java
#	src/main/java/nimbus/tasks/Task.java

jamesberyl committed:
Add JUnit tests for find

jamesberyl committed:
Add checkstyle config

jamesberyl committed:
Edit code style

jamesberyl committed:
Merge branch-A-CheckStyle into master

jamesberyl committed:
Add images for Nimbus and User

jamesberyl committed:
Add fxml files for MainWindow and DialogBox

jamesberyl committed:
Add css files for main and dialog-box

jamesberyl committed:
Update build.gradle to support JavaFX

jamesberyl committed:
Update build.gradle to support JavaFX

jamesberyl committed:
Update build.gradle to support JavaFX

jamesberyl committed:
Fix path to MainWindow

jamesberyl committed:
Create Main class

jamesberyl committed:
Create DialogBox class

jamesberyl committed:
Create Launcher class

jamesberyl committed:
Create MainWindow class

jamesberyl committed:
Modify classes to remove text-based ui and add GUI

jamesberyl committed:
Remove unnecessary code

jamesberyl committed:
Add JavaDocs

jamesberyl committed:
Show welcome message on startup

jamesberyl committed:
Add ability to close app on bye command

jamesberyl committed:
Merge branch-Level-10 into master

weiyanminoo commented on src/main/java/nimbus/DialogBox.java:
```diff
@@ -0,0 +1,83 @@
+     * @param text The text to be displayed in the dialog box.
+     * @param img  The image representing the speaker's face.
+     */
+    private DialogBox(String text, Image img) {
```
Private constructor here might limit extensibility, perhaps can make it protected instead if there are subclasses in the future?

weiyanminoo commented on src/main/java/nimbus/tasks/Deadline.java:
```diff
@@ -0,0 +1,91 @@
+    private LocalDateTime by;
+    private static final DateTimeFormatter OUTPUT_FORMAT = DateTimeFormatter.ofPattern("MMM dd yyyy, h:mm a");
+    private static final List<DateTimeFormatter> INPUT_FORMATS = Arrays.asList(
+            DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm"),
```
i think having different date formats for user input is really good idea but maybe u can consider putting it in a centralized class? since u use it in the Event class too!

weiyanminoo commented on src/main/java/nimbus/ui/UI.java:
```diff
@@ -0,0 +1,155 @@
+ * Responsible for displaying messages, reading user input, and presenting task-related information.
+ */
+public class UI {
+
```
there is an extra line here as well as before the next method

weiyanminoo commented on src/test/java/nimbus/exceptions/NimbusExceptionTest.java:
```diff
@@ -0,0 +1,36 @@
+
+    @Test
+    void testNullMessage() {
+        NimbusException exception = new NimbusException(null);
```
can consider using assertNull() for null checks

weiyanminoo commented on src/test/java/nimbus/tasks/EventTest.java:
```diff
@@ -0,0 +1,43 @@
+    void testIsOnDate() throws NimbusException {
+        Event event = new Event("Workshop", "2023-11-10 0900", "2023-11-11 1700");
+        LocalDateTime date = LocalDateTime.parse("2023-11-11T15:00");
+        assertEquals(true, event.isOnDate(date));
```
Can consider using assertTrue instead

weiyanminoo reviewed (commented):
Hey Beryl! Overall a really impressive job! Your code is very clean and complete and there is thorough testing. I had to really nitpick to find any coding standard issues. Keep it up :))

cscms03 commented on src/main/java/nimbus/tasks/Deadline.java:
```diff
@@ -0,0 +1,91 @@
+ */
+public class Deadline extends Task {
+    private LocalDateTime by;
+    private static final DateTimeFormatter OUTPUT_FORMAT = DateTimeFormatter.ofPattern("MMM dd yyyy, h:mm a");
```
good naming convention for constant formats, strings, methods!

cscms03 commented on src/main/java/nimbus/tasks/Deadline.java:
```diff
@@ -0,0 +1,91 @@
+     * @return The formatted string representing the task for file storage.
+     */
+    @Override
+    public String toFileString() {
```
String.format() could be an option to increase readability!

cscms03 commented on src/main/java/nimbus/tasks/Event.java:
```diff
@@ -0,0 +1,92 @@
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+import java.util.List;
+
```
good import order

cscms03 commented on src/main/java/nimbus/ui/UI.java:
```diff
@@ -0,0 +1,155 @@
+        return "Hey there! I'm Nimbus, your assistant. ☁️\n"
+                + "How can I make your day brighter?";
+    }
+
```
Minor thing: two line breaks between methods

cscms03 reviewed (commented):
Overall, the code is easy to read through as it complies with coding standard. Good quality of code!

jamesberyl committed:
Improve code quality

jamesberyl committed:
Avoid complicated expression

jamesberyl committed:
Improve code quality

jamesberyl committed:
Add assertions in UI

jamesberyl committed:
Add assertions in TaskList

jamesberyl committed:
Enable assertions

jamesberyl committed:
Modify assertions in TaskList

jamesberyl committed:
Add assertions in Main

jamesberyl committed:
Add assertions in DialogBox

jamesberyl committed:
Merge pull request #1 from jamesberyl/branch-A-Assertions

Merge branch-A-Assertions into master through PR

jamesberyl committed:
Merge master into branch-A-CodeQuality

jamesberyl committed:
Merge pull request #2 from jamesberyl/branch-A-CodeQuality

Merge branch-A-CodeQuality through PR

jamesberyl committed:
Implement chronological sorting for tasks

Previously, tasks in Nimbus were just listed in the order they were added, which made it hard to figure out what needed attention first. Deadlines could get lost in the list, and events wouldn’t always show up in the right order.

Without sorting, users had to manually scan through their tasks to see which had the highest priority. This made planning and staying on top of things harder.

Now, tasks are sorted chronologically:
* Events come first, sorted by their start time. If two events start at the same time, they’re ordered by end time.
* Deadlines follow, sorted by due date.
* Todos, which don’t have deadlines, are placed in the order they were created.

This update makes it much easier to see what’s coming up next, helping users stay organised without extra effort.

jamesberyl committed:
Merge branch-C-Sort into master

jamesberyl committed:
Modify boolean naming

jamesberyl committed:
Modify test classes

jamesberyl committed:
Modify main to include name in GUI

jamesberyl committed:
Add ui.png

jamesberyl committed:
Update test classes

jamesberyl committed:
Update .gitignore

jamesberyl committed:
Update UI.java

jamesberyl committed:
Update test classes

jamesberyl committed:
add README.md

jamesberyl committed:
update README.md

jamesberyl committed:
update docs

jamesberyl committed:
update docs

jamesberyl committed:
modify README.md

jamesberyl committed:
Improve code quality

jamesberyl committed:
Fix bug when storing tasks that are marked

jamesberyl committed:
Update output message when adding tasks

jamesberyl committed:
Update test classes

jamesberyl committed:
Update README.md
